domain/src/main/java/org/oppia/android/domain/survey/SurveyGatingController.kt

```diff
@@ -0,0 +1,89 @@
+package org.oppia.android.domain.survey
+
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.domain.exploration.ExplorationActiveTimeController
+import org.oppia.android.domain.profile.ProfileManagementController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders.Companion.transform
+import org.oppia.android.util.locale.OppiaLocale
+import org.oppia.android.util.system.OppiaClock
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+private const val SURVEY_MINIMUM_TOPIC_LEARNING_TIME_MINUTES = 5
+
+private const val GET_SURVEY_LAST_SHOWN_TIMESTAMP_PROVIDER_ID =
+  "get_survey_last_shown_timestamp_provider_id"
+private const val SURVEY_LAST_SHOWN_TIMESTAMP_PROVIDER_ID =
+  "survey_last_shown_timestamp_provider_id"
+private const val GET_TOPIC_LEARNING_TIME_PROVIDER_ID =
+  "get_topic_learning_time_provider_id"
+private const val TOPIC_LEARNING_TIME_PROVIDER_ID =
+  "topic_learning_time_provider_id"
+
+/**
+ * Controller for retrieving survey gating criteria and deciding if a survey should be shown.
+ */
+class SurveyGatingController @Inject constructor(
+  private val profileManagementController: ProfileManagementController,
+  private val oppiaClock: OppiaClock,
+  private val explorationActiveTimeController: ExplorationActiveTimeController,
+  private val machineLocale: OppiaLocale.MachineLocale
+) {
+
+  private val surveyGracePeriodMillis = TimeUnit.DAYS.toMillis(30)
```

Comment:
Is this something we should maybe implement as a platform parameter? It could be useful to be able to remotely tweak it in the future (and platform parameters make it easier to swap out this value while testing).

Reply:
I did find out from testing that platform parameters would be preferrable here.